### PR TITLE
Core Data: Update storage usage in AddOnGroupStore

### PR DESCRIPTION
**Pull request by itsmeichigo**
Part of #14122 

## Description

This PR migrates the write operations for add-on groups and optimizes fetch requests when upserting the groups.

## Steps to reproduce

- Install the WC Product Add-ons plugin to your test store. Since this is a paid plugin, you can also test by uploading the [package](https://github.com/woocommerce/all-plugins/tree/master/product-packages/woocommerce-product-addons) directly to your store.
- Create an order that uses a some add-ons (you can test with both global and non-global ones).
- Launch the app and enable View Add-Ons in Menu > Settings > Experimental Features.
- Navigate to the add-on order and tap the "View Add-Ons" button.
- See that the correct add-ons are displayed

## Testing information

Tested in simulator iPhone 16 Pro iOS 18.1 and confirmed that add-ons are fetched and displayed successfully.

## Screenshots

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Yosemite/Yosemite/Stores/AddOnGroupStore.swift [hunk 1]
```diff
@@ -86,7 +81,10 @@ private extension AddOnGroupStore {
 
         // Delete stale groups
         let activeIDs = readOnlyAddOnGroups.map { $0.groupID }
-        storage.deleteStaleAddOnGroups(siteID: siteID, activeGroupIDs: activeIDs)
+        let staleGroups = storedGroups.filter { !activeIDs.contains($0.groupID) }
+        staleGroups.forEach {
+            storage.deleteObject($0)
+        }
```
Comment (selanthiraiyan) [inline Yosemite/Yosemite/Stores/AddOnGroupStore.swift:87]: Can we remove the code and tests related to the unused `deleteStaleAddOnGroups` method?

**Review by selanthiraiyan — APPROVED** [review]
Nice work, Huong! TIL about the Product Add Ons plugin. 

This tests as expected. I can view the add ons info from the order details screen. 

Code LGTM. I left a comment about removing unused code. 

:shipit:

File: Yosemite/Yosemite/Stores/AddOnGroupStore.swift (hunk 1 above)
Comment (selanthiraiyan) [reply Yosemite/Yosemite/Stores/AddOnGroupStore.swift:87]: Sorry for not being clear. I meant [deleting this method](https://github.com/woocommerce/woocommerce-ios/blob/d25df661be32ee9737abf6af95d34eb9373eab74/Storage/Storage/Tools/StorageType%2BDeletions.swift#L189-L194), which I think isn't needed anymore. What do you think?

Comment (itsmeichigo) [reply Yosemite/Yosemite/Stores/AddOnGroupStore.swift:87]: Nice catch! Done in fcfd916f0e64782810ec78d0f7443ef058a40590.

**Pull request merged**